catalog/core/catalog-core-api/src/main/java/ddf/catalog/filter/SimpleFilterDelegate.java

```diff
@@ -0,0 +1,679 @@
+ */
+public abstract class SimpleFilterDelegate<T> extends FilterDelegate<T> {
+
+    public <S> T defaultOperation(Object property, S literal, Class<S> literalClass,
+            Enum operation) {
+        throw new UnsupportedOperationException(String.format("{} not supported by {} for property",
+                operation,
+                getClass().getName()));
+    }
+
+    public T logicalOperation(Object operand, LogicalPropertyOperation logicalPropertyOperation) {
+        return defaultOperation(operand, null, null, logicalPropertyOperation);
+    }
+
+    public <S> T comparisonOperation(String propertyName, S literal, Class<S> literalClass,
+            ComparisonPropertyOperation comparisonPropertyOperation) {
+        return defaultOperation(propertyName, literal, literalClass, comparisonPropertyOperation);
+    }
+
+    public <S> T spatialOperation(String propertyName, S wkt, Class<S> wktClass,
+            SpatialPropertyOperation spatialPropertyOperation) {
+        return defaultOperation(propertyName, wkt, wktClass, spatialPropertyOperation);
+    }
+
+    public <S> T temporalOperation(String propertyName, S literal, Class<S> literalClass,
+            TemporalPropertyOperation temporalPropertyOperation) {
+        return defaultOperation(propertyName, literal, literalClass, temporalPropertyOperation);
+    }
+
+    public <S> T xpathOperation(String xpath, S literal, Class<S> literalClass,
+            XPathPropertyOperation xpathPropertyOperation) {
+        return defaultOperation(xpath, literal, literalClass, xpathPropertyOperation);
+    }
+
+    // Logical operators
+    public T and(List<T> operands) {
+        return logicalOperation(operands, LogicalPropertyOperation.AND);
+    }
+
+    public T or(List<T> operands) {
+        return logicalOperation(operands, LogicalPropertyOperation.OR);
+    }
+
+    public T not(T operand) {
+        return logicalOperation(operand, LogicalPropertyOperation.NOT);
+    }
+
+    public T include() {
+        return logicalOperation(null, LogicalPropertyOperation.INCLUDE);
+    }
+
+    public T exclude() {
+        return logicalOperation(null, LogicalPropertyOperation.EXCLUDE);
+    }
+
+    // Comparison operators
+
+    // PropertyIsEqualTo
+
+    public T propertyIsEqualTo(String propertyName, String literal, boolean isCaseSensitive) {
+        return propertyIsEqualTo(propertyName,
+                literal,
+                String.class,
+                ComparisonPropertyOperation.IS_EQUAL_TO);
+    }
+
+    public T propertyIsEqualTo(String propertyName, Date literal) {
+        return propertyIsEqualTo(propertyName,
+                literal,
+                Date.class,
+                ComparisonPropertyOperation.IS_EQUAL_TO);
+    }
+
+    public T propertyIsEqualTo(String propertyName, Date startDate, Date endDate) {
+        return propertyIsEqualTo(propertyName,
+                new DateRange(startDate, endDate),
```

Comment:
Instead of adding a Geotools dependency, we might be able to do with with `Range` from Guava.  

https://github.com/google/guava/wiki/RangesExplained